src/main/java/de/eldoria/pickmeup/listener/CarryListener.java

```diff
@@ -50,99 +57,348 @@ public CarryListener(Plugin plugin, Configuration config, ProtectionService prot
+                .ifPresent(MountHandler::dispose);
+        throwBarHandler.getAndRemove(player);
+    }
+
     private enum MountState {
         SNEAK_MOUNT, WALKING, SNEAK_THROW
     }
+
+    private class MountHandler{
+        public MountState state;
+        public BukkitTask throwTask;
+
+        private final Player owner;
+
+        private ArmorStand offsetter;
+
+        public MountHandler(Player owner){
+            this.owner = owner;
+        }
+
+        public void cleanTasks(){
+            if (throwTask != null && !throwTask.isCancelled()) throwTask.cancel();
+        }
+
+        public void dispose(){
+            cleanTasks();
+
+            if (offsetter == null)
+                return;
+
+            for (Entity passenger :
+                    offsetter.getPassengers()) {
+                offsetter.removePassenger(passenger);
+            }
+            offsetter.remove();
+            offsetter = null;
+        }
+
+        public boolean addPassenger(Entity passenger){
+            Entity mountable = getMountable();
+
+            if(mountable.getUniqueId() == passenger.getUniqueId())
+                return false;
+
+            return mountable.addPassenger(passenger);
+        }
+
+        public boolean removePassenger(Entity passenger){
+            Entity mountable = getMountable();
+
+            boolean result = false;
+
+            if(mountable.getUniqueId() != passenger.getUniqueId())
+                 result = mountable.removePassenger(passenger);
+
+            // If there are no more passengers, dispose
+            if(mountable.getPassengers().isEmpty())
+                dispose();
+
+            return result;
+        }
+
+        public List<Entity> getDirectPassengers(){
+            Entity mountable = getMountable();
+
+            return mountable.getPassengers();
+        }
+
+        public List<Entity> getAllPassengers(boolean filterOffsetters){
+            return getAllPassengers(owner, filterOffsetters);
+        }
+        public int stackCount(boolean filterOffsetters){
+            return stackCount(owner, filterOffsetters);
+        }
+
+        private void setupOffsetter() {
+            if (offsetter != null && !offsetter.isDead())
+                return;
+
+            ArmorStand armorStand = (ArmorStand) owner.getWorld().spawnEntity(owner.getLocation(), EntityType.ARMOR_STAND);
+
+            armorStand.setInvulnerable(true);
+            armorStand.setInvisible(true);
+            //armorStand.setMarker(true);
+            armorStand.setSmall(true);
+            armorStand.setBasePlate(false);
+            //armorStand.setLeftArmPose(90);
+            //armorStand.setRightArmPose(90);
+
+
+            // Mark it as an offsetter so it can be easily identified for removal
+            PersistentDataContainer dataContainer = armorStand.getPersistentDataContainer();
+            dataContainer.set(PickMeUp.offsetterIdentifierKey(), PersistentDataType.BYTE, (byte) 1);
```

Comment:
Can be handled by eldo utils

```suggestion
                        DataContainerUtil.putValue(armorStand, PickMeUp.offsetterIdentifierKey(), PersistentDataType.BYTE, (byte) 1);
```